docs/learn/encyclopedia/contract-development/events.mdx

```diff
@@ -133,3 +127,97 @@ The `fn_return` diagnostic event is emitted when a contract call completes and c
 `events` contain `ContractEvents` that should convey information about state changes. `diagnosticEvents` on the other hand contain events that are not useful for most users, but may be helpful in debugging issues or building the contract call stack. Because they won't be used by most users, they can be optionally enabled because they are not hashed into the ledger, and therefore are not part of the protocol. This is done so a stellar-core node can stay in sync with the network while emitting these events that normally would not be useful for most users.
 
 Due to the fact that a node with diagnostic events enabled will be executing code paths that diverge from a regular node, we highly encourage only using this feature on watcher node (nodes where `NODE_IS_VALIDATOR=false` is set).
+
+## Reading events
+
+You can use the [`getEvents`](../../../data/rpc/api-reference/methods/getEvents.mdx) endpoint of any RPC service to fetch and filter events by type, contract, and topic.
+
+:::warning
+
+Events are ephemeral: RPC providers typically only keep short chunks (less than a week) of history around.
+
+:::
+
+To learn more about working with events, take a look at the [events guides](../../../build/guides/events/README.mdx) and [this example contract](../../../build/smart-contracts/example-contracts/events.mdx).
+
+For a quick high-level demonstration, though, we'll use the [TypeScript SDK](../../../tools/sdks/README.mdx) to infinitely fetch all `transfer` events (defined by the [Soroban Token Interface](https://stellar.org/protocol/sep-41#interface)) involving the [XLM contract](https://stellar.expert/explorer/testnet/contract/CDLZFC3SYJYDZT7K67VZ75HPJVIEUVNIXF47ZG2FB2RMQQVU2HHGCYSC) and display them in a human-friendly format.
+
+<CodeExample>
+
+```typescript
+import { humanizeEvents, nativeToScVal } from "@stellar/stellar-sdk";
+import { Server } from "@stellar/stellar-sdk/rpc";
+
+const s = new Server("https://soroban-testnet.stellar.org");
+
+function main() {
+  const r = await s.getLatestLedger();
+
+  const xlmFilter = {
+    type: "contract",
+    contractIds: ["CDLZFC3SYJYDZT7K67VZ75HPJVIEUVNIXF47ZG2FB2RMQQVU2HHGCYSC"],
```

Comment:
are there any helper functions in the typescript sdk to compute the SAC address for a given stellar asset? I think that could be more instructive than hard coding the xlm contract on testnet

Reply:
I did this initially and then undid it for some reason :facepalm: fixed in f074c740